lib/src/rhsm.rs

```diff
@@ -0,0 +1,140 @@
+//! Integration with Red Hat Subscription Manager
+
+use anyhow::Result;
+use cap_std::fs::{Dir, OpenOptions};
+use cap_std_ext::cap_std;
+use fn_error_context::context;
+use serde::Serialize;
+
+const FACTS_PATH: &str = "etc/rhsm/facts/bootc.json";
+
+#[derive(Serialize)]
+struct RhsmFacts {
+    #[serde(rename = "bootc.booted.image")]
+    booted_image: String,
+    #[serde(rename = "bootc.booted.version")]
+    booted_version: String,
+    #[serde(rename = "bootc.booted.digest")]
+    booted_digest: String,
+    #[serde(rename = "bootc.staged.image")]
+    staged_image: String,
+    #[serde(rename = "bootc.staged.version")]
+    staged_version: String,
+    #[serde(rename = "bootc.staged.digest")]
+    staged_digest: String,
+    #[serde(rename = "bootc.rollback.image")]
+    rollback_image: String,
+    #[serde(rename = "bootc.rollback.version")]
+    rollback_version: String,
+    #[serde(rename = "bootc.rollback.digest")]
+    rollback_digest: String,
+    #[serde(rename = "bootc.available.image")]
+    available_image: String,
+    #[serde(rename = "bootc.available.version")]
+    available_version: String,
+    #[serde(rename = "bootc.available.digest")]
+    available_digest: String,
+}
+
+impl From<crate::spec::HostStatus> for RhsmFacts {
```

Comment:
I think we should be able to unit test this pretty easily, see the existing unit tests for the status CLI

Reply:
👍 good thing too because while writing the test I just realized i accidentally swapped version and digest!